src/LinuxTracing/UprobesUnwindingVisitor.cpp

```diff
@@ -759,4 +781,36 @@ void UprobesUnwindingVisitor::Visit(uint64_t event_timestamp, const MmapPerfEven
   listener_->OnModuleUpdate(std::move(module_update_event));
 }
 
+void UprobesUnwindingVisitor::Visit(uint64_t /*event_timestamp*/,
+                                    const CloneExitPerfEventData& event_data) {
+  // If the return value of clone is zero this tracepoint is hit from the execution path of the
+  // newly created thread. We are not interested in these events and discard them.
+  if (event_data.ret_tid == 0) return;
+
+  const pid_t parent_tid = event_data.tid;
+  const pid_t tid_in_target_process_namespace = event_data.ret_tid;
+  if (!new_task_parent_to_root_namespace_tids_.contains(parent_tid)) {
+    ORBIT_ERROR(
+        "Observed a return from clone without previously seeing a task_newtask from the same "
+        "parent thread. parent_tid was %d clone return was value: %d. We will ignore dynamic "
+        "instrumentation form this thread.",
+        parent_tid, tid_in_target_process_namespace);
+    return;
+  }
+  tid_to_root_namespace_tid_[tid_in_target_process_namespace] =
+      new_task_parent_to_root_namespace_tids_[parent_tid];
+  new_task_parent_to_root_namespace_tids_.erase(parent_tid);
+}
+
+void UprobesUnwindingVisitor::Visit(uint64_t /*event_timestamp*/,
+                                    const TaskNewtaskPerfEventData& event_data) {
+  if (new_task_parent_to_root_namespace_tids_.contains(event_data.was_created_by_tid)) {
+    ORBIT_ERROR(
+        "Observed a task_newtask event from thread %d without matching clone exit event. This "
+        "should never happen.",
+        event_data.was_created_by_tid);
+  }
+  new_task_parent_to_root_namespace_tids_[event_data.was_created_by_tid] = event_data.new_tid;
```

Comment:
So `was_created_by_tid` is in the namespace of the target while `new_tid` is in the root namespace?

Reply:
Oh yeah, so see comment on renaming `new_task_parent_to_root_namespace_tids_`.

Reply:
done